### ISSUE TITLE
Publishing APK in beta track causes 403 Error in validation step

### ISSUE THREAD

**Issue by nicolas-fricke** [bitrise-steplib/steps-google-play-deploy#15]
Not sure if issue might be related to issue https://github.com/bitrise-steplib/steps-google-play-deploy/issues/12.

I have two workflows set up, `alpha` and `beta` which get triggered by commits on different Git branches (`develop` and `master` respectively). Both build the APK and then use the `google-play-deploy` step  (v 1.1.0) to publish the APK to the _Play Store_ onto the respective tracks. 

The `versionCode` of the app is set to reflect the Bitrise build number (and therefore auto-increments with each build).

The current version uploaded in the `alpha` track is _81_ and version _63_ is active in the beta track. A push to the `master` branch now triggered a `beta` deploy, Bitrise build number (and therefore also Android `versionCode`) _82_.

This now caused a failure of the step:
```
Failed to validate edit, error: googleapi: Error 403: Version 81 of this app can not be downloaded by any devices as they will all receive APKs with higher version codes., multiApkShadowedActiveApk
```

But from what I understand the behavior in the play store should be that if a higher version is pushed to the `beta` track than what is active in `alpha`, it should deactivate the `alpha` track and those testers should also receive the beta version.

From https://support.google.com/googleplay/android-developer/answer/3131213 :
> Alpha testing group users are eligible for the alpha, beta, or production version of your app (with the highest version code that's compatible with their device). Beta testing group users are only eligible for the beta or production version of your app (with the highest version code that's compatible with their device).
> 
> This means your alpha APKs should have the highest version codes, followed by beta, and finally production.
> 
> - If a beta APK is uploaded with a higher version code than an alpha APK, alpha users who can install both will install the beta APK instead of the alpha APK.
> - If a production APK is uploaded with a higher version code than an alpha or beta APK, some or all of your alpha and beta users will install the production APK, instead of APKs of the testing tracks. 

Full console output of the failed step:

```
+------------------------------------------------------------------------------+
| (0) google-play-deploy@1.1.0                                                 |
+------------------------------------------------------------------------------+
| id: google-play-deploy                                                       |
| version: 1.1.0                                                               |
| collection: https://github.com/bitrise-io/bitrise-steplib.git                |
| toolkit: go                                                                  |
| time: 2017-02-20T15:38:24Z                                                   |
+------------------------------------------------------------------------------+
|                                                                              |
INFO[15:38:24] Start installing (golang) with apt-get       
INFO[15:38:24]  * [OK] Step dependency (go) installed, available. 

Configs:
- JSONKeyPath: *****
- PackageName: *****
- ApkPath: /bitrise/deploy/app-release-unsigned-bitrise-signed.apk
- Track: beta
- UserFraction: 0.5
- WhatsnewsDir: 
Deprecated Configs:
- ServiceAccountEmail: 
- P12KeyPath: 

Authenticateing
Authenticated client created

Create new edit
 editID: *****

Upload apks
 uploaded apk version: 82

Update track
 updated track: beta
 assigned apk versions: [82]

Validating edit
Failed to validate edit, error: googleapi: Error 403: Version 81 of this app can not be downloaded by any devices as they will all receive APKs with higher version codes., multiApkShadowedActiveApk
|                                                                              |
+---+---------------------------------------------------------------+----------+
| x | google-play-deploy@1.1.0 (exit code: 1)                       | 9.50 sec |
+---+---------------------------------------------------------------+----------+
| Issue tracker: https://github.com/bitrise-io/steps-google-play-deploy/issues |
| Source: https://github.com/bitrise-io/steps-google-play-deploy               |
+---+---------------------------------------------------------------+----------+
```

**Comment by nicolas-fricke** [comment]
It seems like other projects had similar issues with the Google Developer Console API and found a work-around: https://github.com/Triple-T/gradle-play-publisher/issues/121

**Comment by godrei** [comment]
Hi @nicolas-fricke ,

So:
the latest version on __alpha__ was: `81`
the latest version on __beta__ was: `63`

Then you tried to upload 82 to `beta`.

I found a similar [conversation](https://github.com/Triple-T/gradle-play-publisher/issues/121), where they explains the same situation as yours. (you can not upload higher version apk to beta track then the highest one on alpha).

But i can not reproduce this locally (nor using the web interface neither using the step)[e.g.: i can upload higher version to beta, then the highest on alpha].

In our test app, we use closed testing on both alpha and beta tracks.

What kind of testing do you use on your tracks?

When i upload a higher version on beta, the alpa apk is deactivated automatically. 

What kind of permissions did you provide to your service account?

**Comment by nicolas-fricke** [comment]
Hi @godrei, thanks for getting back :)

We have set up both beta and alpha tracks to be _"Testing using Google Groups or Google+ Communities"_.

For me, uploading a higher versioned APK to beta manually to the Developer Console works flawlessly, it deactivates the alpha track and then publishes the higher versioned beta app. But whenever the step runs on Bitrise and the alpha track is still active (with a lower versioned APK), then the step fails.

When the alpha track has been previously disabled by manually uploading a beta version, then the step succeeds in uploading a higher versioned beta version, too. But once there's a new alpha version published, the step fails again.

The linked Google Account has the following permissions within the Developer Console:
 - Visibility
 - Edit store listing, pricing & distribution
 - Manage Production APKs
 - Manage Alpha & Beta APKs
 - Manage Alpha & Beta users

**Comment by bhurling** [comment]
This is exactly how we did it in the gradle-play-publisher: See [this file](https://github.com/Triple-T/gradle-play-publisher/blob/170a7e878811026f3d5ea175870f6883694079de/src/main/groovy/de/triplet/gradle/play/PlayPublishApkTask.groovy#L46).

Basically, when uploading something to beta, we untrack every version from alpha with a lower version number. When uploading to production we untrack everything from alpha and beta that is blocking. But we moved that feature behind a special flag in the config so we do not accidentally deactivate APKs.
